src/swift/Block.swift

```diff
@@ -104,8 +104,8 @@ public class DispatchWorkItem {
 /// uses @convention(block). This is to avoid unnecessary bridging between
 /// C blocks and Swift closures, which interferes with dispatch APIs that depend
 /// on the referential identity of a block. Particularly, dispatch_block_create.
-internal typealias _DispatchBlock = @convention(block) () -> Void
-internal typealias dispatch_block_t = @convention(block) () -> Void
+internal typealias _DispatchBlock = @escaping @convention(block) () -> Void
+internal typealias dispatch_block_t = @escaping @convention(block) () -> Void
```

Comment:
I get your point, but for what I understand that is a change that should concern the C part of libdispatch. Today `dispatch_block_t` is being declared in `object.h` like this:

```
typedef void (^dispatch_block_t)(void);
```

and `internal typealias dispatch_block_t` doesn't seem to be more than a redeclaration of the existing C type. `dispatch_block_t` is being used both in `dispatch_sync` and `dispatch_async` for that matter, that means that it can be both `@escaping` and `@noescape`, and from both options `@escaping` is not the correct, but the safe option.

A different approach may be having a `dispatch_noescape_block_t` with `__attribute__((noescape))`, but that will create a difference with the darwin API.

For what I understand here, `internal typealias dispatch_block_t` is just briging the C type `oid (^dispatch_block_t)(void)` to the Swift scope, and it's being only used internally.

Am I getting this wrong? What would be the correct solution given the context?

Thanks.